cores/nRF5/Tone.cpp

```diff
@@ -34,161 +34,397 @@ Version Modified By Date     Comments
 0010	Arduino.org 16/07/27 Added Arduino Primo support
 *************************************************/
 
-
+#include "Arduino.h"
 #include "Tone.h"
 #include "WVariant.h"
+#include "limits.h"
+
+// NOTE: Currently hard-coded to only use PWM2 ...
+//       These are the relvant hard-coded variables
+// TODO: Consider allowing dynamic use of any available PWM peripheral (but start with #2 for compatibility)
+static NRF_PWM_Type * const _PWMInstance = NRF_PWM2;
+static HardwarePWM  * const _HwPWM       = HwPWMx[2];
+
+// Defined a struct, to simplify validation testing ... also provides context when debugging
+class TONE_PWM_CONFIG {
+    private:
+        const nrf_pwm_clk_t      clock_frequency = NRF_PWM_CLK_125kHz;
+        const nrf_pwm_mode_t     pwm_mode        = NRF_PWM_MODE_UP;
+        const nrf_pwm_dec_step_t step_mode       = NRF_PWM_STEP_AUTO;
+        const nrf_pwm_dec_load_t load_mode       = NRF_PWM_LOAD_COMMON;
+        const uintptr_t          toneToken       = 0x656e6f54; //< 'T' 'o' 'n' 'e'
+    private:
+        uint64_t pulse_count;         //< total number of PWM pulses
+        uint32_t seq0_refresh;        //< count of pulses for each SEQ0 iteration
+        uint32_t seq1_refresh;        //< count of pulses for each SEQ1 iteration
+        uint16_t loop_count;          //< how many times to restart SEQ0/SEQ1?
+        uint16_t time_period;         //< how many clock cycles allocated to each PWM pulse?
+        uint16_t duty_with_polarity;  //< SEQ[N].PTR will point here, length == 1
+        uint8_t arduino_pin;          //< the arduino pin for playback
```

Comment:
we should only store either arduino_pin or nrf_pin not both.

Reply:
That's quite an assertion.  Why "should" this only store one?

Clearly, we "should" store both because it makes debugging easier, storing the nrf pin allows simpler checking of peripheral config, and storing the arduino pin number helps confirm the user pin  (without having to manually peruse the variant's mapping table).  Oh, and it takes zero extra space.

I think I'll leave this one as-is.